Source/stores.cpp

```diff
@@ -1699,25 +1683,28 @@ void WitchSellEnter()
 	stextvhold = stextsval;
 
 	int idx = stextsval + ((stextsel - stextup) / 4);
-	Players[MyPlayerId].HoldItem = storehold[idx];
-	if (StoreGoldFit(idx))
-		StartStore(STORE_CONFIRM);
-	else
+
+	if (!StoreGoldFit(storehold[idx])) {
 		StartStore(STORE_NOROOM);
+		return;
+	}
+
+	StoreItem = storehold[idx];
+	StartStore(STORE_CONFIRM);
 }
 
 /**
  * @brief Recharges an item in the player's inventory or body in the witch.
  */
-void WitchRechargeItem()
+void WitchRechargeItem(int price)
 {
-	auto &myPlayer = Players[MyPlayerId];
-
-	TakePlrsMoney(myPlayer.HoldItem._iIvalue);
+	TakePlrsMoney(price);

```

Comment:
I get what you are saying, but as things are it's not super clear to me when the items value is manipulated to hold a temporary store exchange value versus the actual items value. I feel that I would be running a risk of introducing bugs by making assumptions regarding `StoreItem == storehold[idx]` at this point and would rather do so when addressing the whole `stextlhold`, `stextsel`, `stextvhold`, `stextsval`, `storehold` mess. Also `BoyBuyItem()` does not operate with `idx` and this way they get to have a similar interface.

The idea behind having Repair/Recharge/Identify take the value was that in the future the shop prices would be separate from `item._iIvalue` since manipulating the items makes things fragile and confusing. But I can make them take `Item &item` as the input instead if that makes a difference to you.

Reply:
> I get what you are saying, but as things are it's not super clear to me when the items value is manipulated to hold a temporary store exchange value versus the actual items value.

The _ivalue/_iIvalue variables in storehold are always the cost of action/vendor offer (either 1/4 the item's known value or the cost for repairing/recharging/identifying the item). Your point about avoiding introducing bugs is good though, I brought it up here because of the added parameter but it's easy enough to change later.

> The idea behind having Repair/Recharge/Identify take the value was that in the future the shop prices would be separate from `item._iIvalue` since manipulating the items makes things fragile and confusing. But I can make them take `Item &item` as the input instead if that makes a difference to you.

I would definitely not want this to take an item reference as the sole param 😂 . What you've got now is good enough, functions the same (at least from my limited testing) and solves the original purpose.